components/plugin_marketplace/marketplace_item/marketplace_item_plugin/marketplace_item_plugin.tsx

```diff
@@ -284,10 +295,63 @@ export default class MarketplaceItemPlugin extends React.PureComponent <Marketpl
         this.props.actions.installPlugin(this.props.id, this.props.version);
     }
 
+    getPluginStatusButton(): React.ReactNode {
+        if (!this.props.installedVersion) {
+            return null;
+        }
+
+        const pluginEnabled = this.props.pluginStatus?.state === PluginState.PLUGIN_STATE_RUNNING || this.props.pluginStatus?.state === PluginState.PLUGIN_STATE_STARTING || this.props.pluginStatus?.state === PluginState.PLUGIN_STATE_FAILED_TO_START;
+        let switchDisabled = this.props.pluginStatus?.state !== PluginState.PLUGIN_STATE_RUNNING && this.props.pluginStatus?.state !== PluginState.PLUGIN_STATE_NOT_RUNNING && this.props.pluginStatus?.state !== PluginState.PLUGIN_STATE_FAILED_TO_START;
+
+        const limit = this.props.limits?.integrations?.enabled;
+        const usage = this.props.integrationsUsage?.enabled;
+        const ignored = Constants.Integrations.FREEMIUM_USAGE_IGNORED_PLUGINS.includes(this.props.id);
+        if (!ignored && !pluginEnabled && limit && usage && usage >= limit) {
+            switchDisabled = true;
+        }
+
+        const label = pluginEnabled ? (
+            <FormattedMessage
+                id='marketplace_modal.plugin_status.enabled'
+                defaultMessage='Enabled'
+            />
+        ) : (
+            <FormattedMessage
+                id='marketplace_modal.plugin_status.disabled'
+                defaultMessage='Disabled'
+            />
+        );
+
+        return (
+            <>
+                <Toggle
+                    id={`plugin-enable-toggle-${this.props.id}`}
+                    disabled={switchDisabled}
+                    onToggle={() => {
+                        if (pluginEnabled) {
+                            this.props.actions.disablePlugin(this.props.id);
+                        } else {
+                            this.props.actions.enablePlugin(this.props.id);
+                        }
+                    }}
+                    toggled={pluginEnabled}
+                    className='btn-sm'
+                />
+                <span className='plugin-status-label'>
+                    {label}
+                </span>
+            </>
+        );
+    }
+
     getItemButton(): JSX.Element {
+        let actionButton: React.ReactNode;
+
         if (this.props.installedVersion !== '' && !this.props.installing && !this.props.error) {
-            return (
+            actionButton = (
                 <Link
+
+                    // site url not supported here
                     to={'/admin_console/plugins/plugin_' + this.props.id}
```

Comment:
Accidentally left this comment in here. This needs to use site url to support servers using a subpath